### PR TITLE
Map 1292 update an inactive cell

### PR DESCRIPTION
**Pull request by BarryGee-MOJ**
The changes in this PR are related to the [MAP-1292 - Update an inactive cell](https://dsdmoj.atlassian.net/jira/software/c/projects/MAP/boards/1354?selectedIssue=MAP-1292) ticket.

### CODE REVIEW

File: cypress.config.ts [hunk 1]
```diff
@@ -30,6 +30,7 @@ export default defineConfig({
     baseUrl: 'http://localhost:3007',
     excludeSpecPattern: '**/!(*.cy).ts',
     specPattern: 'integration_tests/e2e/**/*.cy.{js,jsx,ts,tsx}',
+    // specPattern: 'integration_tests/e2e/changeTemporaryDeactivationsDetails/*.cy.{js,jsx,ts,tsx}',
```
Comment (Mjwillis) [inline cypress.config.ts:33]: Should this be removed?

**Review by Mjwillis — COMMENTED** [review]
remove commented out?

File: cypress.config.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (BarryGee-MOJ) [reply cypress.config.ts]: No, but it has been removed and pushed.

File: server/controllers/changeTemporaryDeactivationDetails/details.test.ts [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+          value: 'Description',
+        },
+        'deactivationReasonDescription-TEST2': {
+          ...fields.deactivationReasonDescription,
+          id: 'deactivationReasonDescription-TEST2',
+          name: 'deactivationReasonDescription-TEST2',
+        },
+      })
+    })
+  })
```
[the remark this answers is not in the document]
Comment (jimbali) [reply server/controllers/changeTemporaryDeactivationDetails/details.test.ts:202]: Otherwise lgtm

[the remark this answers is not in the document]
Comment (BarryGee-MOJ) [reply server/controllers/changeTemporaryDeactivationDetails/details.test.ts:220]: additional tests added and pushed for saveValues and successHandler.

File: server/controllers/changeTemporaryDeactivationDetails/details.ts [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+        ] as string,
+
+        estimatedReactivationDate: estimatedReactivationDate as string,
+        planetFmReference: planetFmReference as string,
+      }
+
+      const valuesHaveChanged = this.compareInitialAndSubmittedValues({
+        initialValues: this.getInitialValues(req, res),
+        submittedValues,
+      })
```
Comment (jimbali) [inline server/controllers/changeTemporaryDeactivationDetails/details.ts:116]: A lot of the other journeys have what I think is the same logic, where it checks if the values have changed and if not then it returns to the view location page, but it's implemented in a different way here:
https://github.com/ministryofjustice/hmpps-locations-inside-prison/blob/main/server/controllers/changeCellCapacity/index.ts#L49-L60

Not a big issue but something to be aware of maybe

**Review by jimbali — APPROVED** [review]

**Pull request merged**